src/esockd_proxy_proto.erl

```diff
@@ -106,21 +120,74 @@ parse_v2(?LOCAL, _Trans, _ProxyInfo, #proxy_socket{socket = Sock}) ->
 
 parse_v2(?PROXY, ?STREAM, ProxyInfo, ProxySock = #proxy_socket{inet = inet4}) ->
     <<A:8, B:8, C:8, D:8, W:8, X:8, Y:8, Z:8,
-      SrcPort:16, DstPort:16, _/binary>> = ProxyInfo,
-    {ok, ProxySock#proxy_socket{src_addr = {A, B, C, D}, src_port = SrcPort,
-                                dst_addr = {W, X, Y, Z}, dst_port = DstPort}};
+      SrcPort:16, DstPort:16, AdditionalFields/binary>> = ProxyInfo,
+    parse_v2_additional(AdditionalFields, ProxySock#proxy_socket{
+        src_addr = {A, B, C, D}, src_port = SrcPort,
+        dst_addr = {W, X, Y, Z}, dst_port = DstPort});
 
 parse_v2(?PROXY, ?STREAM, ProxyInfo, ProxySock = #proxy_socket{inet = inet6}) ->
     <<A:16, B:16, C:16, D:16, E:16, F:16, G:16, H:16,
       R:16, S:16, T:16, U:16, V:16, W:16, X:16, Y:16,
-      SrcPort:16, DstPort:16, _/binary>> = ProxyInfo,
-    {ok, ProxySock#proxy_socket{src_addr = {A, B, C, D, E, F, G, H}, src_port = SrcPort,
-                                dst_addr = {R, S, T, U, V, W, X, Y}, dst_port = DstPort}};
+      SrcPort:16, DstPort:16, AdditionalFields/binary>> = ProxyInfo,
+    parse_v2_additional(AdditionalFields, ProxySock#proxy_socket{
+        src_addr = {A, B, C, D, E, F, G, H}, src_port = SrcPort,
+        dst_addr = {R, S, T, U, V, W, X, Y}, dst_port = DstPort});
 
 parse_v2(_, _, _, #proxy_socket{socket = Sock}) ->
     esockd_transport:fast_close(Sock),
     {error, unsupported_proto_v2}.
 
+
+parse_v2_additional(<<>>, ProxySock) ->
+    {ok, ProxySock};
+parse_v2_additional(<<?PP2_TYPE_ALPN:8, Len:16, PP2_ALPN:Len/binary, Remainder/bitstring>>, ProxySock) ->
+    parse_v2_additional(Remainder, ProxySock#proxy_socket{pp2_alpn = PP2_ALPN});
+parse_v2_additional(<<?PP2_TYPE_AUTHORITY:8, Len:16, PP2_AUTHORITY:Len/binary, Remainder/bitstring>>, ProxySock) ->
+    parse_v2_additional(Remainder, ProxySock#proxy_socket{pp2_authority = PP2_AUTHORITY});
+parse_v2_additional(<<?PP2_TYPE_CRC32C:8, Len:16, PP2_CRC32C:Len/binary, Remainder/bitstring>>, ProxySock) ->
+    parse_v2_additional(Remainder, ProxySock#proxy_socket{pp2_crc32c = PP2_CRC32C});
+parse_v2_additional(<<?PP2_TYPE_NOOP:8, Len:16, _Ignore:Len/binary, Remainder/bitstring>>, ProxySock) ->
+    parse_v2_additional(Remainder, ProxySock);
+parse_v2_additional(<<?PP2_TYPE_NETNS:8, Len:16, PP2_NETNS:Len/binary, Remainder/bitstring>>, ProxySock) ->
+    parse_v2_additional(Remainder, ProxySock#proxy_socket{pp2_netns = PP2_NETNS});
+parse_v2_additional(<<?PP2_TYPE_SSL:8, Len:16, PP2_SSL:Len/binary, Remainder/bitstring>>, ProxySock) ->
+    {ok, ProxySockWithSSL} = parse_v2_ssl(PP2_SSL, ProxySock),
+    parse_v2_additional(Remainder, ProxySockWithSSL);
+parse_v2_additional(InvalidAdditionalFields, _ProxySock) ->
+    {error, {invalid_pp2_additional_fields, InvalidAdditionalFields}}.
+
+
+parse_v2_ssl(<<_Unused:5, PP2_CLIENT_SSL:1, PP2_CLIENT_CERT_CONN:1, PP2_CLIENT_CERT_SESS:1,
```

Comment:
<<_Unused:5, PP2_CLIENT_SSL:1, PP2_CLIENT_CERT_CONN:1, PP2_CLIENT_CERT_SESS:1

should be

<<_Unused:5, PP2_CLIENT_CERT_SESS:1, PP2_CLIENT_SSL:1, PP2_CLIENT_CERT_CONN:1